### PR TITLE
copy session attachments

### PR DESCRIPTION
**Pull request by dpark6060**
Successful run: With attached files, excluded dicom:
https://ss.ce.flywheel.io/#/jobslog/job/5f22f8e4763881003d588f41

Successful run: With attached files, excluded dicom, copied file metadata:
https://ss.ce.flywheel.io/#/jobslog/job/5f22fae57638810039588f43

Successful run: Didn’t copy files because “export_session_attachments” was false:
https://ss.ce.flywheel.io/#/jobslog/job/5f22fc017638810041588eb2

### CODE REVIEW

File: run.py [hunk 1]
```diff
@@ -47,6 +47,63 @@ def false_if_exc_timeout_or_sub_exists(exception):
...
+    if len(session_files) == 0:
+        log.info('No files to copy over')
+        return
+
+    for session_file in session_files:
+        
+        if session_file.type == 'dicom':
+            log.warning(f"File {session_file.name} is type DICOM.\n"
+                        f"DICOMS Uploaded as attachments to a session will NOT be migrated,\n"
+                        f"As we cannot properly ensure de-identification at this level.\n"
```
Comment (lmperry) [inline run.py:77]:
```suggestion
                        f"As we do not support DICOM mapping at this level.\n"
```
This gear does not do de-id.

File: manifest.json [hunk 2]
```diff
@@ -2,11 +2,11 @@
   "name": "session-export",
   "label": "GRP-9: Session Export",
   "description": "Export data (including metadata) from a given session to the specified 'export_project'. The gear will also read DICOM header information from Flywheel metadata and modify DICOM headers to reflect the changes made. Optionally, original data can be 'archived' to an <archive_project>, as configured during the gear execution. The exported, and optionally archived, session will be tagged as appropriate using the 'EXPORTED' tag. At present this gear can only be run at the session level. Output is an export log in csv format.",
-  "version": "1.3.2",
+  "version": "1.3.3",
```
Comment (lmperry) [inline manifest.json:5]:
```suggestion
  "version": "1.4.0",
```

File: manifest.json [hunk 3]
```diff
@@ -2,11 +2,11 @@
...
   "custom": {
     "gear-builder": {
       "category": "analysis",
-      "image": "flywheel/session-export:1.3.2"
+      "image": "flywheel/session-export:1.3.3"
```
Comment (lmperry) [inline manifest.json:9]:
```suggestion
      "image": "flywheel/session-export:1.4.0"
```

File: run.py [hunk 4]
```diff
@@ -47,6 +47,63 @@ def false_if_exc_timeout_or_sub_exists(exception):
 ###############################################################################
 # LOCAL FUNCTION DEFINITIONS
 
+def _copy_files_from_session(fw, from_session, to_session):
+    """Copies file attachments from one session to another
```
Comment (lmperry) [inline run.py:51]:
```suggestion
    """Exports file attachments from one session to another. DICOM files will not be exported.
```

File: run.py [hunk 5]
```diff
@@ -47,6 +47,63 @@ def false_if_exc_timeout_or_sub_exists(exception):
...
+
+        
+        download_file = os.path.join('/tmp', session_file.name)
+        log.debug(f"\tdownloading file {session_file.name} to {download_file}")
+        from_session.download_file(session_file.name, download_file)
+        log.debug("\tcomplete")
+
+        log.debug("\tuploading file to new session")
+        try:
+            metad = {"info": session_file.info}
```
Comment (kalebdfischer) [inline run.py]: Needs classification, modality and type copied if the values are truthy. The classification should be treated the same as acquisition classifications (pop empty lists, validate the classification, etc). I'd also break out preparing the file metadata string into a function of its own.

**Pull request merged**